### DISCUSSION TITLE
Question about the radiation pattern from focal mechanism

### DISCUSSION THREAD

**Discussion by wangzb0329** [SeismicSource/sourcespec#17] [Q&A] [answered]
Hi, 
I would like to use the focal mechanism of earthquakes to correct the radiation pattern of each station, but I don't have the QuakeML file to save the focal mechanism results.
Is there another way to input the focal mechanism results, or could you add an option to input the value of strike/dip/rake angle in the conf file？
Looking forward to your kind advice

**Comment by wangzb0329** [comment, marked as answer]
This problem has been solved. I have created the QuakeML file and use the focal mechanism, thanks😄

**Reply by claudiodsf** [reply]
Hi, sorry for late reply... I'm glad you could solve your problem by creating a QuakeML file (ObsPy is a good tool for that).

I will consider adding in the future a way of specifying earthquake focal parameters (origin, mechanism) through a simple text file.

**Reply by woxin5295** [reply]
Dear Prof. Satriano
Do I still need to make QuakeML files in order to use FM? My input file is just the SAC file. I am not good at using M files. I want to use the seismic moment tensor.

**Reply by claudiodsf** [reply]
Dear Junqing,

Yes, for the moment only QuakeML input is supported for providing the focal mechanism.

What's the format of your FM file? Could you please send me an example? I will see whether it is easy to add support for it.

**Reply by woxin5295** [reply]
My No06 event, whose MT format is as follows, and the two lines below correspond to each other.

Mrr      Mtt    Mpp     Mrt    Mrp    Mtp     exponent(Nm)
0.290  0.091 -0.381  0.238 -2.073 -0.473    19

**Reply by woxin5295** [reply]
Dear Prof. Satriano, I have one more question, and that is， whether SourceSpec would be possible to fix Mo (or Mw) value, in order to prevent tradeoff with fc, because we KNOW the Mo value very well and are interested primarily in fc.

Junqing

**Reply by claudiodsf** [reply]
> My No06 event, whose MT format is as follows, and the two lines below correspond to each other.
> 
> Mrr Mtt Mpp Mrt Mrp Mtp exponent(Nm)
> 0.290 0.091 -0.381 0.238 -2.073 -0.473 19

Is this format the output of a specific program? It looks like GMT psmeca except that the exponent is N-m instead of dyne-cm

**Reply by claudiodsf** [reply]
> whether SourceSpec would be possible to fix Mo (or Mw) value, in order to prevent tradeoff with fc, because we KNOW the Mo value very well and are interested primarily in fc.

Not currently, but it is something I can consider including.

**Reply by woxin5295** [reply]
> > My No06 event, whose MT format is as follows, and the two lines below correspond to each other.
> > Mrr Mtt Mpp Mrt Mrp Mtp exponent(Nm)
> > 0.290 0.091 -0.381 0.238 -2.073 -0.473 19
> 
> Is this format the output of a specific program? It looks like GMT psmeca except that the exponent is N-m instead of dyne-cm

It is from the ISOLA, which is a seismic moment tensor inversion software developed Jiří Zahradník.
